Raven.Database/Actions/MaintenanceActions.cs

```diff
@@ -242,5 +250,113 @@ public void DeleteRemovedIndexes(Dictionary<int, DocumentDatabase.IndexFailDetai
                 }
             }, TaskCreationOptions.LongRunning);
         }
+
+        public void StartDeletingIndexDataAsync(int indexId, string indexName)
+        {
+            pendingDeletions.Enqueue(new DeleteIndexTask
+            {
+                IndexId = indexId,
+                IndexName = indexName
+            });
+
+            if (Interlocked.CompareExchange(ref lockStatus, Locked, UnLocked) == Locked)
```

Comment:
Bad, race condition with the deletion task. You can get to a point where you are just before the finally setting it to unlock, but this code thinks that the deletion task it going to handle it.